ui/components-react/src/components-react/tree/controlled/TreeModel.ts

```diff
@@ -517,3 +525,28 @@ export function getVisibleDescendants(
 
   return result;
 }
+
+function cloneSubree(source: TreeModel, target: MutableTreeModel, parentId: string | undefined): void {
+  target.setNumChildren(
+    parentId,
+    (parentId === undefined ? source.getRootNode() : source.getNode(parentId)!).numChildren,
+  );
+  for (const [childId, index] of source.getChildren(parentId)?.iterateValues() ?? []) {
+    const node = source.getNode(childId)!;
+    target.setChildren(
+      parentId,
+      [{
+        description: node.description,
+        isExpanded: node.isExpanded,
+        id: node.id,
+        item: node.item,
+        label: node.label,
+        isLoading: !!node.isLoading,
```

Comment:
Since this is a cloning operation, is there a reason to force the type of `isLoading` to boolean ? If it ever change type, not forcing it to boolean here would not require a change to this function in the future.

In the same idea, using a spread operation would ensure that if properties are ever added to the node object, the cloning operation would keep working. (Unless there is a reason not to do so.)

Reply:
We need to keep a cast for `isLoading` to boolean due to unfortunate interface mismatch between the types involved.

I see how current cloning implementation creates opportunities for programming errors. Implemented your suggestion in be329fcc1d09aa5b05998d0698097c2e6a2f5e10.